### DISCUSSION TITLE
Error message: ERROR: Underlying core creation failed while creating collection

### DISCUSSION THREAD

**Discussion by ls0218** [irgroup-classrooms/dis17-2021#35] [Q&A] [answered]
Hello,

my solr instance was working without any problems, but since two days I am getting an error message saying "ERROR: Failed to create collection 'cord19_211229' due to: Underlying core creation failed while creating collection: cord19_211229". Sometimes I cannot even start solr. I already tried to reinstall solr but it didn't help. I think it has something to do with zookeeper because I am getting the following warning message when I start solr:
```
INFO  - 2021-12-31 09:23:29.918; org.apache.solr.common.cloud.ConnectionManager; Waiting for client to connect to ZooKeeper
INFO  - 2021-12-31 09:23:29.938; org.apache.solr.common.cloud.ConnectionManager; zkClient has connected
INFO  - 2021-12-31 09:23:29.939; org.apache.solr.common.cloud.ConnectionManager; Client is connected to ZooKeeper
INFO  - 2021-12-31 09:23:29.957; org.apache.solr.common.cloud.ZkStateReader; Updated live nodes from ZooKeeper... (0) -> (2)
INFO  - 2021-12-31 09:23:29.968; org.apache.solr.client.solrj.impl.ZkClientClusterStateProvider; Cluster at localhost:9983 ready
WARN  - 2021-12-31 09:23:29.972; org.apache.zookeeper.ClientCnxn; An exception was thrown while closing send thread for session 0x100001d74730003. => EndOfStreamException: Unable to read additional data from server sessionid 0x100001d74730003, likely server has closed socket
        at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x100001d74730003, likely server has closed socket
        at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77) ~[zookeeper-3.6.2.jar:3.6.2]
        at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350) ~[zookeeper-3.6.2.jar:3.6.2]
        at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1275) [zookeeper-3.6.2.jar:3.6.2]
```

Does somebody else have this problem or an idea how to solve it?

Thank you!

**Comment by ls0218** [comment]
We figured out that the error messages occures when we change the managed schema to the schema.xml in the default configset (actually changing something in the default configset is not a good idea in general). We have problems uploading our own configset as well as creating a new collection using our own configuration. We need to do that because we want to change the ranking method. 

Does any other group also struggle with this problem?

**Reply by ghost** [reply]
hi leonie, we had the same problem, and it might be connected to working in the cloud. you would need to upload your configsets to zookeeper, see https://solr.apache.org/guide/8_11/using-zookeeper-to-manage-configuration-files.html.
alternatively, we tried running a standalone instance which worked quite well. see here for more: https://solr.apache.org/guide/8_11/solr-configuration-files.html
hope this helps!

**Reply by ls0218** [reply]
Hi Helga! Thank you very much for your answer. It is working now.

**Comment by phschaer** [comment, marked as answer]
Dear all, 

the Zookeeper is part of "Solr Cloud", which enables Solr to be run in a virtual cloud environment. According to the Solr documentation: 

> Apache Solr includes the ability to set up a cluster of Solr servers that combines fault tolerance and high availability. Called SolrCloud, these capabilities provide distributed indexing and search capabilities, supporting the following features:
> * Central configuration for the entire cluster
> * Automatic load balancing and fail-over for queries
> * ZooKeeper integration for cluster coordination and configuration.


While this is a great feature, it's not useful in our setting. 

__How to get around this problem__: Start Solr with `.bin/solr start` instead of `./bin/solr start -e cloud` 

My guess here is, that you started your Solr installation with the latter (like you might have done in the very first tutorial), right?

**Reply by ls0218** [reply]
Dear Mr. Schaer, 

thank you very much. That was our mistake. Starting it with .bin/solr start solved the problem.

Kind regards
Leonie Schulz

**Comment by ls0218** [comment]
Dear Mr. Breuer,
thank you very much for your help today. It worked the way you told us and it's actually pretty easy to do it that way. 
@breuert

**Reply by breuert** [reply]
Great news! I am happy that it works now!
